src/components/diagrams/diagram-pane.js

```diff
@@ -286,95 +362,314 @@ export function DiagramPane({
+            // We check if we need to add new diagrams
+            const diagramsToAdd = [];
+            diagramStates.forEach((diagramState) => {
+                if (diagramState.svgType !== DiagramType.NETWORK_AREA_DIAGRAM) {
+                    const diagramAlreadyPresentInViews = viewsRef.current.find(
+                        (diagramView) =>
+                            diagramView.svgType !==
+                                DiagramType.NETWORK_AREA_DIAGRAM &&
+                            diagramView.id === diagramState.id
+                    );
+                    if (!diagramAlreadyPresentInViews) {
+                        diagramsToAdd.push({
+                            ...diagramState,
+                            name: intl.formatMessage(
+                                { id: 'LoadingOf' },
+                                { value: diagramState.id }
+                            ),
+                            align: 'left',
+                            loadingState: true,
+                        });
+                    }
+                }
+            });
+
+            // If we found diagrams to add, we add them
+            if (diagramsToAdd?.length) {
+                // First we add the empty diagrams in the views
+                setViews((views) => {
+                    const updatedViews = views
+                        .concat(diagramsToAdd)
+                        .sort(sortByAlign);
+                    return updatedViews;
+                });
 
-                    diagramViews.push({
-                        ...emptyDiagramView,
-                        align: 'left',
+                // Then we add the data when the fetch is finished
+                diagramsToAdd.forEach((diagramState) => {
+                    createView(diagramState).then((singleLineDiagramView) => {
+                        setViews((views) => {
+                            const diagramViewId = views.findIndex(
+                                (view) =>
+                                    view.svgType !==
+                                        DiagramType.NETWORK_AREA_DIAGRAM &&
+                                    view.id === diagramState.id
+                            );
+                            const updatedViews = views.slice();
+                            // we update the SLD with the fetched data
+                            updatedViews[diagramViewId] = {
+                                ...updatedViews[diagramViewId],
+                                ...singleLineDiagramView,
+                                loadingState: false,
+                            };
+                            return updatedViews;
+                        });
                     });
+                });
+            }
+        },
+        [createView, intl]
+    );
+
+    // Check if we need to remove old SLDs from the 'views' and remove them if necessary
+    const removeSLDs = useCallback((diagramStates) => {
```

Comment:
Maybe rename this function to "removeObsoleteSLDs" ?

Reply:
Then I should rename "addSLDs" in "addNewSLDs" ?